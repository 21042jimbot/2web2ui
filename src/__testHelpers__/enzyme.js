export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
