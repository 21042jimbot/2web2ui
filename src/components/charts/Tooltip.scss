// A very dangerous and global classname
.LiftTooltip {
  position: relative;

  .recharts-tooltip-wrapper {
    z-index: 1;
  }
}
