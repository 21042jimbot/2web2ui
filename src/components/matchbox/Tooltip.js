import React from 'react';
import { Tooltip as OGTooltip } from '@sparkpost/matchbox';
import { Tooltip as HibanaTooltip } from '@sparkpost/matchbox-hibana';
import { useHibana } from 'src/context/HibanaContext';
import { omitSystemProps } from 'src/helpers/hibana';

export default function Tooltip(props) {
  const [state] = useHibana();
  const { isHibanaEnabled } = state;

  if (!isHibanaEnabled) {
    return <OGTooltip {...omitSystemProps(props, ['color'])} />;
  }
  return <HibanaTooltip {...props} />;
}

OGTooltip.displayName = 'OGTooltip';
HibanaTooltip.displayName = 'HibanaTooltip';
