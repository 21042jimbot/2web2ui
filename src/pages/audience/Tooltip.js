import React from 'react';
import { Box, Stack } from '@sparkpost/matchbox-hibana';
import { format } from 'date-fns';
import { getFriendlyFilterLabel } from './Filters';

function Tooltip(props) {
  const { data } = props;
  const friendlyLabel = getFriendlyFilterLabel(data);

  return (
    <Box bg="gray.900" p="200" color="white" borderRadius="200" boxShadow="200" width="200px">
      <Stack space="100">
        <Box fontSize="200">{format(data.date, 'MMM DD YYYY')}</Box>
        <Box borderLeft="3px solid rgba(235, 240, 245, 1)" pl="300">
          <Box fontSize="200" color="gray.400">
            Delivered
          </Box>
          <Box fontSize="300">{Number(data.delivery).toLocaleString()}</Box>
        </Box>
        <Box borderLeft="3px solid #78b6ff" pl="300">
          <Box fontSize="200" color="gray.400">
            {friendlyLabel}
          </Box>
          <Box fontSize="300">
            {((data.value / data.delivery) * 100).toFixed(1)}% â€“ {Number(data.value).toLocaleString()}
          </Box>
        </Box>
      </Stack>
    </Box>
  );
}

export default Tooltip;
