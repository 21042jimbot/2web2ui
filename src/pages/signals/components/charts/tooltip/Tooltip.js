import React from 'react';
import { formatDate } from 'src/helpers/date';
import styles from './Tooltip.module.scss';
import './Tooltip.scss';

const Tooltip = ({ children, date }) => (
  <div className={styles.TooltipWrapper}>
    {date && (
      <div className={styles.TooltipDate}>
        {formatDate(date)}
      </div>
    )}
    <div className={styles.TooltipContent}>
      {children}
    </div>
  </div>
);

export default Tooltip;
