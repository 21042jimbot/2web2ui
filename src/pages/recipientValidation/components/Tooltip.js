import React from 'react';
import { InfoOutline } from '@sparkpost/matchbox-icons';
import { Tooltip } from 'src/components/matchbox';
import styles from './Tooltip.module.scss';

const InfoTooltip = ({ content }) => (
  <Tooltip
    children={<InfoOutline className={styles.TooltipIcon} size={16} />}
    content={<div className={styles.content}>{content}</div>}
    dark
    horizontalOffset="-.8rem"
  />
);

export default InfoTooltip;
