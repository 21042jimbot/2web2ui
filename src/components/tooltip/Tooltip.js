import React, { useState } from 'react';
import { Popover } from 'src/components/matchbox';
import useUniqueId from 'src/hooks/useUniqueId';
import style from './Tooltip.module.scss';

// note, Matchbox's Tooltip has static dimensions that prevent us from doing fun things
// todo, should use Matchbox's Tooltip
const Tooltip = ({ children, content, initialOpen = false }) => {
  const [isOpen, setOpen] = useState(initialOpen);
  const uniqueId = useUniqueId('tooltip-');

  return (
    <Popover
      id={uniqueId}
      children={content}
      className={style.Content}
      open={isOpen}
      sectioned={false}
      trigger={
        <span
          aria-controls={uniqueId}
          onMouseEnter={() => {
            setOpen(true);
          }}
          onMouseLeave={() => {
            setOpen(false);
          }}
        >
          {children}
        </span>
      }
    />
  );
};

export default Tooltip;
