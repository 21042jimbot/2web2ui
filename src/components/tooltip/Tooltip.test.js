import React from 'react';
import { shallow } from 'enzyme';
import Tooltip from './Tooltip';

describe('Tooltip', () => {
  const subject = (props = {}) => shallow(
    <Tooltip
      children={<button>Learn More</button>}
      content={<div>More</div>}
      {...props}
    />
  );

  it('renders correctly', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('opens content when hovered', () => {
    const wrapper = subject();
    const triggerWrapper = shallow(wrapper.prop('trigger'));

    triggerWrapper.simulate('mouseenter');

    expect(wrapper).toHaveProp('open', true);
  });

  it('closes content when mouse leaves trigger', () => {
    const wrapper = subject({ initialOpen: true });
    const triggerWrapper = shallow(wrapper.prop('trigger'));

    triggerWrapper.simulate('mouseleave');

    expect(wrapper).toHaveProp('open', false);
  });
});
