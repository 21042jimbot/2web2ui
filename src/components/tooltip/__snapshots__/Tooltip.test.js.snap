// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<Popover
  bottom={true}
  className="Content"
  open={false}
  right={true}
  sectioned={false}
  trigger={
    <span
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button>
        Learn More
      </button>
    </span>
  }
>
  <div>
    More
  </div>
</Popover>
`;
